y3prediction/prediction.py

```diff
@@ -1487,87 +1645,76 @@ def check_boundary(boundary, name):
             return False
 
         if rank == 0:
-            print(f"Found boundary {name} in fluid domain")
+            print(f"Found boundary named {name} in fluid domain")
         return True
 
-    if use_outflow_boundary:
-        use_outflow_boundary = check_boundary(outflow_bnd, "outflow")
-    if use_inflow_boundary:
-        use_inflow_boundary = check_boundary(inflow_bnd, "inflow")
-    if use_flow_boundary:
-        use_flow_boundary = check_boundary(flow_bnd, "flow")
-    if use_injection_boundary:
-        use_injection_boundary = check_boundary(inj_bnd, "injection")
-    if use_wall_boundary:
-        use_wall_boundary = check_boundary(wall_bnd, "wall")
-    if use_interface_boundary:
-        use_interface_boundary = check_boundary(interface_bnd, "interface")
-
-    if (use_outflow_boundary and use_flow_boundary or
-            use_inflow_boundary and use_flow_boundary):
-        error_message = \
-            "Invalid boundary configuration, inflow/outflow with flow:"
-        from mirgecom.simutil import SimulationConfigurationError
-        raise SimulationConfigurationError(error_message)
+    # setup element boundary assignments
+    bndry_elements = {}
+    for bnd_name in bndry_config:
+        # skip disabled boundaries
+        if bndry_config[bnd_name] != "none":
+            # check to see if any elements are assigned to this named boundary,
+            # if not, disabled it
+            bndry_elements[bnd_name] = dd_vol_fluid.trace(bnd_name)
+            bnd_exists = check_boundary(bndry_elements[bnd_name], bnd_name)
+            if not bnd_exists:
+                bndry_config[bnd_name] = "none"
```

Comment:
Not sure this will work right with domain decomp, won't some boundaries be limited to subsets of the partitions?